packages/plugin-ext/src/main/browser/workspace-main.ts

```diff
@@ -152,22 +152,28 @@ export class WorkspaceMainImpl implements WorkspaceMain {
         });
     }
 
-    async $startFileSearch(includePattern: string, excludePatternOrDisregardExcludes?: string | false,
-        maxResults?: number, token?: theia.CancellationToken): Promise<UriComponents[]> {
-        const opts: FileSearchService.Options = { rootUris: this.roots.map(r => r.uri) };
+    async $startFileSearch(includePattern: string, includeFolderUri: string | undefined, excludePatternOrDisregardExcludes?: string | false,
+        maxResults?: number, theiaToken?: theia.CancellationToken): Promise<UriComponents[]> {
```

Comment:
@benoitf @evidolob Is cancellation hooked into plugin RCP? from https://github.com/theia-ide/theia/pull/4549#issuecomment-472839390 it looks like not. Is there another way to cancel?

Reply:
fyi, this is what vscode does from the extension host:

```ts
		if (token && token.isCancellationRequested) {
			return Promise.resolve([]);
		}

		return this._proxy.$startFileSearch(includePattern, includeFolder, excludePatternOrDisregardExcludes, maxResults, token)
			.then(data => Array.isArray(data) ? data.map(URI.revive) : []);
```

Reply:
i know, but it does not seem to work in Theia, i would need to check